apps/dotcms-ui/src/app/view/components/dot-edit-layout-designer/dot-edit-layout-designer.component.ts

```diff
@@ -71,25 +75,45 @@ export class DotEditLayoutDesignerComponent implements OnInit, OnDestroy, OnChan
     form: FormGroup;
     initialFormValue: any;
     themeDialogVisibility = false;
+    dialogDisplay = false;
+    dialogHeader: string;
+    unSaved = true;
     currentTheme: DotTheme;
 
     saveAsTemplate: boolean;
     showTemplateLayoutSelectionDialog = false;
 
     private destroy$: Subject<boolean> = new Subject<boolean>();
 
+    @HostListener('window:beforeunload', ['$event'])
+    saveChangesOnUnload() {
+        if(!_.isEqual(this.form.value, this.initialFormValue)){
+            this.onSave();
+            this.cd.detectChanges();
+        }
+        return true;
+    }
+
     constructor(
         private dotEditLayoutService: DotEditLayoutService,
         private dotEventsService: DotEventsService,
         private dotHttpErrorManagerService: DotHttpErrorManagerService,
         private dotRouterService: DotRouterService,
         private dotThemesService: DotThemesService,
+        private dotGlobalMessageService: DotGlobalMessageService,
+        private dotMessageService: DotMessageService,
+        private canDesactiveGuardService: CanDesactiveGuardService,
```

Comment:
Is a bad practice to use a guard as a service, you need to create an intermediate service that guard and component can use to share data between them.